database/dbutil.go

```diff
@@ -140,12 +140,12 @@ func PersistNamespacesAndCommit(datastore Datastore, namespaces []Namespace) err
 // rollback.
 func FindAncestryAndRollback(datastore Datastore, name string) (Ancestry, bool, error) {
 	tx, err := datastore.Begin()
-	defer tx.Rollback()
-
 	if err != nil {
 		return Ancestry{}, false, err
 	}
 
+	defer tx.Rollback()
```

Comment:
Hey there! Great working tracking this down; this is a super common mistake in Go.

I only have one tiny nitpick to fix which is to put the defer statement in the same block as the error check:

```go
tx, err := datastore.Begin()
if err != nil {
  return Ancestry{}, false, err
}
defer tx.Rollback()

return tx.FindAncestry(name)
```

Reply:
No problem with nitpicks :-) It takes a while to get the feel for others coding styles etc.

(I had to choose between the func above and the func below to determine where to put the defer and I chose the wrong one !)